sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -1070,6 +1070,62 @@ void ProgramManager::addImages(pi_device_binaries DeviceBinary) {
         ArgMaskMap[Info->Name] =
             createKernelArgMask(pi::DeviceBinaryProperty(Info).asByteArray());
     }
+
+    // Fill maps for kernel bundles
+    if (EntriesB != EntriesE) {
+      std::lock_guard<std::mutex> KernelIDsGuard(m_KernelIDsMutex);
+
+      // Register all exported symbols
+      auto ExportedSymbols = Img->getExportedSymbols();
+      for (const pi_device_binary_property &ExportedSymbol : ExportedSymbols)
+        m_ExportedSymbols.insert(ExportedSymbol->Name);
+
+      m_BinImg2KernelIDs[Img.get()].reset(new std::vector<kernel_id>);
+
+      for (_pi_offload_entry EntriesIt = EntriesB; EntriesIt != EntriesE;
+           ++EntriesIt) {
+
+        // Skip creating unique kernel ID if it is a service kernel.
+        // SYCL service kernels are identified by having
+        // __sycl_service_kernel__ in the mangled name, primarily as part of
+        // the namespace of the name type.
+        if (std::strstr(EntriesIt->name, "__sycl_service_kernel__")) {
+          m_ServiceKernels.insert(EntriesIt->name);
+          continue;
+        }
+
+        // Skip creating unique kernel ID if it is an exported device
+        // function. Exported device functions appear in the offload entries
+        // among kernels, but are identifiable by being listed in properties.
+        if (m_ExportedSymbols.find(EntriesIt->name) != m_ExportedSymbols.end())
+          continue;
+
+        // ... and create a unique kernel ID for the entry
+        auto It = m_KernelName2KernelIDs.find(EntriesIt->name);
+        if (It == m_KernelName2KernelIDs.end()) {
+          std::shared_ptr<detail::kernel_id_impl> KernelIDImpl =
+              std::make_shared<detail::kernel_id_impl>(EntriesIt->name);
+          sycl::kernel_id KernelID =
+              detail::createSyclObjFromImpl<sycl::kernel_id>(KernelIDImpl);
+
+          It = m_KernelName2KernelIDs.emplace_hint(It, EntriesIt->name,
+                                                   KernelID);
+        }
+
+        m_KernelName2KernelIDs.insert(
+            std::make_pair(EntriesIt->name, It->second));
```

Comment:
I am a little confused by this piece. Doesn't the above code emplace the entry into `m_KernelName2KernelIDs` if it isn't there already?

Reply:
You are right, this line is not needed(leftover from latest modifications), will remove.